definitions/scenarios/self_upgrade.rb

```diff
@@ -21,7 +21,9 @@ def current_version
     end
 
     def maintenance_repo_id(version)
-      return ENV['maintenance_repo'] unless ENV['maintenance_repo'].empty?
+      unless ENV['maintenance_repo'].nil?
+        return ENV['maintenance_repo'] unless ENV['maintenance_repo'].empty?
+      end
```

Comment:
Could you do this as:

```
return ENV['maintenance_repo'] if ENV['maintenance_repo']
```

Reply:
No because `.empty?` it's still good to check for a non-empty string. In Rails and Ruby 2.5 you can do `var&.present?` but this isn't rails (so `.present?` doesn't exist) and we don't have the safe operator due to Ruby 2.0.

I'd write this as:
```ruby
if (repo = ENV['maintenance_repo'])
  return repo unless repo.empty?
end
```